### PR TITLE
Chemical requests console feature

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[FEATURE]
## About the PR 
Remake of #9361 with all map changes removed for being cursed. I will make a separate PR to add it to maps once this is merged.
No changelog for this PR, since it doesn't actually add the machines to maps

Adds a chemical request console for placing orders with chemistry. One is found outside science and one in medbay on every map. (I had to move/remove a couple of things to make them fit, actual mappers feel free to tell me if it looks wrong)
Might be good to give botany one as well, lmk what you think.

Request interface:
image image

There is a corresponding console in chemistry itself for fulfilling or denying requests, and a PDA alert is sent to all science PDAs when a request is placed.
image
image

Both consoles currently use the same sprite ala QM consoles and because I am not creative: 
image

## Why's this needed? 
Chemistry in theory is able to quickly make and distribute chemicals for the crew via the mail chutes, however this rarely happens as radio and PDA messages tend to be missed and chemistry is often isolated from the rest of the station. (kudos to donut 3, kondaru and clarion for having chemistry front desks)
Adding a formal request system should make it a lot easier for people to request what they want, and hopefully encourage chemistry to fulfil those requests at some point.
The chemistry terminal keeps a history of all requests made and whether or not they were denied, allowing for some investigation into who is to blame for the clown getting access to thermite.

### CODE REVIEW

**Comment by Momoberry** [comment]
JOB DUTY FOR SCIENCE, WOOOOOOOO

**Comment by TobleroneSwordfish** [comment]
!merge_upstream

**Comment by TobleroneSwordfish** [comment]
merge driver be kind 🙏

**Comment by TobleroneSwordfish** [comment]
!merge_upstream

**Comment by TobleroneSwordfish** [comment]
!merge_upstream

File: tgui/packages/tgui/interfaces/ChemRequester.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ */
+
+import { useBackend, useLocalState } from '../backend';
+import { Button, Input, Section, Box, Stack, NumberInput, LabeledList } from '../components';
+import { Window } from '../layouts';
+import { KEY_ENTER } from 'common/keycodes';
+import { capitalize } from '../../common/string';
+import { ListSearch } from './common/ListSearch';
+
+const IDCard = (props, context) => {
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/ChemRequester.js:15]: I'd like this in Common, actually

File: code/obj/machinery/computer/chem_requester.dm [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+				src.request.note = strip_html(copytext(params["notes"], 1, 66))
+			if ("set_volume")
+				src.request.volume = clamp(params["volume"], 1, src.max_volume)
+				. = TRUE
+			if ("submit")
+				src.request.area_name = src.area_name || (get_area(src))?.name || src.request.area_name
+				src.request.requester_name = src.card.registered + " ([src.card.assignment])"
+				src.request.time = ticker.round_elapsed_ticks
+				//byond jank, lists are only associative if they aren't int indexed
+				chem_requests["[src.request.id]"] = src.request
```
Comment (ZeWaka) [inline code/obj/machinery/computer/chem_requester.dm:79]: might be cooler to store this on the mainframe or something, packets so nerds can sniff and see?
(not required, idea)

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/machinery/computer/chem_requester.dm [hunk 3]
```diff
@@ -0,0 +1,152 @@
+var/list/datum/chem_request/chem_requests = list()
```
Comment (ZeWaka) [inline code/obj/machinery/computer/chem_requester.dm:1]: i'd like the file moved to the chemistry module folder, other than that all looks good

**Review by ZeWaka — APPROVED** [review]

**Comment by TobleroneSwordfish** [comment]
Okay I've done the review changes, I want to do packets for this at some point but that is likely going to have to be after I refactor `machinery/networked` for the sake of my sanity so should probably be in a separate PR.

**Pull request merged**